geode-core/src/main/java/org/apache/geode/cache/query/internal/CompiledIndexOperation.java

```diff
@@ -104,7 +105,13 @@ public Object evaluate(ExecutionContext context) throws TypeMismatchException,
     }
 
     if (rcvr instanceof Map) {
-      return ((Map) rcvr).get(index);
+      if (context instanceof QueryExecutionContext) {
+        return ((Map) rcvr).get(index);
+      }
+      if (((Map<?, ?>) rcvr).containsKey(index)) {
```

Comment:
Both if statements do the same thing, maybe add an || to the previous if call, but I am ok with keeping them separate.